packages/cli/src/cli/basemaps/action.export.ts

```diff
@@ -0,0 +1,180 @@
+                tilesetMap.set(item.name, (recs = new Map())); // new name
+            }
+
+            let curr = recs.get(item.projection);
+            if (curr == null) {
+                curr = [null, null];
+                recs.set(item.projection, curr); // new projection
+            }
+            curr[index] = item; // fill slot
+        } else if (table.Imagery.recordIsImagery(item)) {
+            imageryNames.set(item.id, item.name); // collect imagery names
+        }
+    }
+
+    const codes: Record<string, ProjectionConfig> = {};
+
+    const projections = tilesetMap.get('aerial');
+
+    if (projections != null) {
+        for (const [proj, [head, prod]] of projections.entries()) {
+            const rules = new Map<string, ImageryConfig>();
+
+            if (head != null && (prod == null || head.version !== prod.version)) {
+                for (const rule of head.rules) {
+                    rules.set(rule.ruleId, toImageryConfig(head.id, rule, false));
+                }
+            }
+
+            if (prod != null) {
+                for (const rule of prod.rules) {
+                    const curr = rules.get(rule.ruleId);
+                    if (curr == null) {
+                        rules.set(rule.ruleId, toImageryConfig(prod.id, rule, true));
+                    } else if (
+                        curr.priority == rule.priority &&
+                        curr.zoom[0] == rule.minZoom &&
+                        curr.zoom[1] == rule.maxZoom
+                    ) {
+                        curr.production = true;
+                    }
+                }
+            }
+
+            codes['EPSG:' + proj] = {
+                headBackground: rgbaToHex(head?.background ?? DefaultBackground),
+                prodBackground: rgbaToHex(prod?.background ?? DefaultBackground),
+                imagery: Array.from(rules.values()).sort(compareNamePriority),
+            };
+        }
+    }
+
+    return { aerial: codes };
+}
+
+export class ExportAction extends CommandLineAction {
+    private output: CommandLineStringParameter;
+    private full: CommandLineFlagParameter;
+
+    public constructor() {
+        super({
+            actionName: 'export',
+            summary: 'Export imagery and tileset rules',
+            documentation: '',
+        });
+    }
+
+    protected onDefineParameters(): void {
+        this.output = this.defineStringParameter({
+            argumentName: 'JSON_FILE',
+            parameterLongName: '--output',
+            parameterShortName: '-o',
+            description: 'File to output to',
+            required: true,
+        });
+        this.full = this.defineFlagParameter({
+            parameterLongName: '--full',
+            description: 'Export the entire tilemetadatdb',
+            required: false,
+        });
+    }
+
+    protected async onExecute(): Promise<void> {
+        const outFile = this.output.value!;
+
+        try {
+            await fs.stat(outFile);
```

Comment:
stat then write is not a good idea as it causes race conditions, your better off doing a writeFile with exception if it exists 

`flag: "wx"`

Reply:
Cool that is what I was looking for :)